protocol/messenger.go

```diff
@@ -1337,6 +1339,32 @@ func (m *Messenger) watchExpiredMessages() {
 	}()
 }
 
+// watchUnmutedChats regularly checks for chats that should be unmuted
+func (m *Messenger) watchUnmutedChats() {
+	m.logger.Debug("watching unmuted chats")
+	go func() {
+		for {
+			select {
+			case <-time.After(3 * time.Second): // Poll every 3 seconds
+				chats, _ := m.persistence.Chats()
+				for _, chat := range chats {
+					if time.Now().Unix() > chat.MuteTill && chat.MuteTill != 0 && chat.Muted {
+						_ = m.persistence.UnmuteChat(chat.ID)
+						response := &MessengerResponse{}
```

Comment:
You want to have the response outside the `chats` loop, inside the chat loop you do `response.AddChat(chat)`
and after the loop you dispatch the signal.

So instead of sending 1 signal per chat, you send a single singnal.
Also you want to check if there's anything to dispatch if you do that, since we don't want to send empty signals